org.omg.sysml/src/org/omg/sysml/generation/MOF2SysMLText.xtend

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * SysML 2 Pilot Implementation
+ * Copyright (c) 2022 Model Driven Solutions, Inc.
+ *    
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *  
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  
+ * @license LGPL-3.0-or-later <http://spdx.org/licenses/LGPL-3.0-or-later>
+ *  
+ *******************************************************************************/
+
+package org.omg.sysml.generation
+
+import org.omg.sysml.generation.MOF2KerMLText
+import org.eclipse.uml2.uml.DataType
+import org.eclipse.uml2.uml.Model
+import org.eclipse.uml2.uml.Enumeration
+
+class MOF2SysMLText extends MOF2KerMLText {
+	
+	override generate(Model model) {
+		'''
+		package SysML {
+			private import ScalarValues::*;
+			import KerML::Kernel::*;
+			
+			«model.getPackage("Systems").toPackageBody»
+			
+		}
+		'''
+	}
+	
+	override toEnumeration(Enumeration enumeration) {
+		'''
+		enum def «nameOf(enumeration)» {
+			«FOR literal: enumeration.ownedLiterals»
+				enum «nameOf(literal)»;
+			«ENDFOR»
+		}
+		'''
+	}
+	
+	override toMetaclass(org.eclipse.uml2.uml.Class class_) {
+		toMetaclass(class_, "metadata def")
+	}
+		
+	override toFeature(org.eclipse.uml2.uml.Property property) {
+		if (property.type instanceof DataType) {
+			toFeature(property, false, "attribute")
+		} else {
+			toFeature(property, false, if (property.isComposite) "item" else "ref item")
```

Comment:
The `metadata` keyword (in SysML) declares a MetadataUsage, which is both an ItemUsage and a MetadataFeature, and the latter is a kind of AnnotatingElement. So, `metadata` always introduces a metadata annotation. For a feature that is not an annotation (which is what is needed in the reflective model), it is necessary to treat the metadata as an item. (This is also why 'feature' is used rather than 'metadata' in the KerML model.)